pai-management/k8sPaiLibrary/maintainlib/common.py

```diff
@@ -188,28 +188,40 @@ def ssh_shell_paramiko(host_config, commandline):
 
 
 def ssh_shell_paramiko_with_result(host_config, commandline):
+
     hostip = str(host_config['hostip'])
     if ipv4_address_validation(hostip) == False:
         return False
+
     username = str(host_config['username'])
     password = str(host_config['password'])
     port = 22
     if 'sshport' in host_config:
         if port_validation(host_config['sshport']) == False:
             return (None, None)
         port = int(host_config['sshport'])
+
     ssh = paramiko.SSHClient()
     ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
     ssh.connect(hostname=hostip, port=port, username=username, password=password)
     stdin, stdout, stderr = ssh.exec_command(commandline, get_pty=True)
+    stdin.write(password + '\n')
+    stdin.flush()
     logger.info("Executing the command on host [{0}]: {1}".format(hostip, commandline))
     result_stdout = ""
     for response_msg in stdout:
+        res_msg = response_msg.replace('\n', '')
+        res_msg = res_msg.replace('\r', '')
+        if password == res_msg:
+            # TODO: We should change this behavior.
```

Comment:
@hwuu About this comment, what's your opinion?

Reply:
We can simply ignore the first line of the stdout, since it is definitely the content of stdin. And by the way, shall we require that the 'username' have password-less sudo privilege?

Reply:
Is it a known behavior ?

Reply:
Known behavior, at least according to my own test. We need to do more extensive tests after all. I mean, we may assume that the user does not need to input the password when doing sudo on target machines. Is it a rational requirement?

Reply:
- Fixed by filter the first line.
- But in some situation, user will have to input the password. At least, I come across this issue.